esp-hal/src/dma/mod.rs

```diff
@@ -1878,6 +1878,37 @@ where
     }
 }
 
+/// Holds all the information needed to configure a DMA channel for a transfer.
+pub struct Preparation {
+    pub(crate) start: *mut DmaDescriptor,
+    pub(crate) length: usize,
+    // burst_mode, alignment, check_owner, etc.
+}
+
+/// [DmaTxBuffer] is a DMA descriptor + memory combo that can be used for
+/// transmitting data from a DMA channel to a peripheral's FIFO.
+pub trait DmaTxBuffer {
```

Comment:
Because the users can't create `Preparation` structs, these new traits are essentially sealed, right?

Reply:
I suppose so. (Users could create wrappers around the existing bufs I guess)

I don't intend to seal the new traits, I expect users to eventually create implementations of their own, I just hadn't settled on a `Preparation` constructor yet.